cpp/src/copying/copy.cu

```diff
@@ -241,43 +241,14 @@ std::unique_ptr<column> scatter_gather_based_if_else(Left const& lhs,
   //   1. Struct scalars are not yet available.
   //   2. List scalars do not yet support explosion to a full column.
   CUDF_FAIL("Scalars of nested types are not currently supported!");
+  (void)lhs;
```

Comment:
Did you look into using `[[maybe_unused]]` (C++17) instead of doing this?

Reply:
I had only checked if `[[maybe_unused]]` had any effect when provided on the method arguments it self, but `nvcc` ignores the attribute when provided on method arguments.
```cpp
void function( [[maybe_unused]] std::string& argument)
```

When I try to use `[[maybe_unused]]` on the variables usage we get compile warnings as the compiler belives the statement has no effect. I presume that this is a side effect of `nvcc` not fully supporting this attribute:

```
../../src/copying/copy.cu:244:18: error: statement has no effect [-Werror=unused-value]
  244 |   [[maybe_unused]]lhs;

../../src/copying/copy.cu:244:18: error: statement has no effect [-Werror=unused-value]
  244 |   [[maybe_unused]] lhs;    
```